apps/cowswap-frontend/src/modules/orders/hooks/useSWRProdOrders.ts

```diff
@@ -12,18 +12,19 @@ import { getOrders } from 'api/gnosisProtocol'
 import { useApiOrders } from './useApiOrders'
 import { useSWROrdersRequest } from './useSWROrdersRequest'
 
+const EMPTY_ORDERS: EnrichedOrder[] = []
+
 // Fetch PROD orders only when current env is not prod
 // We need them for TWAP, because WatchTower creates orders only on Prod
 export function useSWRProdOrders(): EnrichedOrder[] {
   const { chainId } = useWalletInfo()
   const requestParams = useSWROrdersRequest()
   const apiOrders = useApiOrders()
 
-  const { data: loadedProdOrders = [] } = useSWR<EnrichedOrder[]>(
+  const { data: loadedProdOrders = EMPTY_ORDERS } = useSWR<EnrichedOrder[]>(
```

Comment:
ok, great, we return always the same empty object, so in case there's no orders, the depending components and hooks won't re-render cause is the same identity (i assume the issue was the fact `[] !== []`)

My only concern is that this might be covering up the real place where the issue originated

Isn't the issue that we are re-rendering somewhere everything even if the loader orders doesn't change?

Reply:
>i assume the issue was the fact [] !== []

Exactly.

>Isn't the issue that we are re-rendering somewhere everything even if the loader orders doesn't change?

I've revisited all hook deps and found problem only here